### ISSUE TITLE
[Feature Request] Add support for ftp on non standard ports

### ISSUE THREAD

**Issue by azureblaze** [ncw/rclone#1434]
Currently the new ftp interface doesn't seem to support ftp servers that run on ports other than the standard port 21, support for other ports would be nice to have.

**Comment by sjurtf** [comment]
The FTP client rclone uses support handling ports in the URL string.
So if you specify the port as part of the URL, the client should be able to connect to non-standard ports.

Like this:
```
FTP URL
url> ftp://my.ftphost.com:5555
Remote config
--------------------
[ftp]
username = sjurtf
password = *** ENCRYPTED ***
url = ftp://my.ftphost.com:5555
--------------------
```

I just tested it, and it works fine.

**Comment by ncw** [comment]
Anyone fancy updating [the docs](https://github.com/ncw/rclone/blob/master/docs/content/ftp.md) with a bit more help about ports?

**Comment by sjurtf** [comment]
I updated docs in PR #1437.

**Comment by sjurtf** [comment]
@ncw, I thought about this one more time. 

Wouldn't it make sense that the FTP and SFTP remotes handle destination URLs the same way during setup of a new remote?

Now the FTP fs requires a custom port to be specified in the same input field as the remote host address, but the SFTP fs has an own input option when a remote is created.

We should be consistent about this. Which way do you prefer?

**Comment by azureblaze** [comment]
Ah, I see, I didn't include the ftp scheme in the URI before, seems to work
pretty well now

**Comment by ncw** [comment]
@sjurtf thanks for your PR I have merged that :-)

> Wouldn't it make sense that the FTP and SFTP remotes handle destination URLs the same way during setup of a new remote?

Yes possibly and given that SFTP has been released and FTP hasn't now is the time to change the FTP if we want to.

Supplying a URL lets you supply a default directory to log in to for FTP.

We could add this to sftp which could be useful, though it is actually a bit of a pain to code as rclone has its own idea of root directory when you do `ftp:/home/user` so I'd be tempted to remove that from the ftp client.

I think it would probably be easier for the users to get a Host, Port, User, Password like the sftp rather than have to construct an FTP URI which as this issue demonstrates isn't familiar to most people.

**Comment by easy90rider** [comment]
> Supplying a URL lets you supply a default directory to log in to for FTP.

If my path has spaces, do I need to add escape chars?

**Comment by ncw** [comment]
@easy90rider you'll need to put the path on the rclone command line now in the latest beta. Yes you will need to use usual shell escaping rules for spaces.

**Comment by ncw** [comment]
This issue was fixed in #1441 so I'm going to close it now.
